salt/tests/unit/modules/files/test_metalk8s_kubernetes_utils.yaml

```diff
@@ -0,0 +1,96 @@
+  - api_server_pillar: {
+      "kubeconfig": "my-pillar-kubeconfig.conf",
+      "context": "my-pillar-context"
+    }
+    config_options: null
+    kwargs: {}
+    result: ['my-pillar-kubeconfig.conf', 'my-pillar-context']
+  # 1. get kubeconfig from salt config options only
+  - api_server_pillar: null
+    config_options: {
+      "kubernetes.kubeconfig": "my-config-option-kubeconfig.conf",
+      "kubernetes.context": "my-config-option-mycontext"
+    }
+    kwargs: {}
+    result: ['my-config-option-kubeconfig.conf', 'my-config-option-mycontext']
+  # 2. get kubeconfig from kwargs options only
+  - api_server_pillar: null
+    config_options: null
+    kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf", "context": "my-kwargs-mycontext"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 3. get kubeconfig partly from api-server-pillars and salt config options
+  - api_server_pillar: {"kubeconfig": "my-pillar-kubeconfig.conf"}
+    config_options: {"kubernetes.context": "my-config-option-mycontext"}
+    kwargs: {}
+    result: ['my-pillar-kubeconfig.conf', 'my-config-option-mycontext']
+  # 4. get kubeconfig partly from kwargs and salt config options
+  - api_server_pillar: null
+    config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    kwargs: {"context": "my-kwargs-mycontext"}
+    result: ['my-config-option-kubeconfig.conf', 'my-kwargs-mycontext']
+  # 5. get kubeconfig partly from kwargs and api-server pillars
+  - api_server_pillar: {"context": "my-pillar-context"}
+    config_options: null
+    kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf"}
+    result: ['my-kwargs-kubeconfig.conf', 'my-pillar-context']
+  # 6. get kubeconfig partly from salt config options and context result is None
+  - api_server_pillar: null
+    config_options: {"kubernetes.kubeconfig": "my-config-option-kubeconfig.conf"}
+    kwargs: {}
+    result: ['my-config-option-kubeconfig.conf', null]
+  # 7. get kubeconfig partly from api-server pillar and context result is None
+  - api_server_pillar: {"kubeconfig": "my-pillar-kubeconfig.conf"}
+    config_options: null
+    kwargs: {}
+    result: ['my-pillar-kubeconfig.conf', null]
+  # 8. get kubeconfig partly from kwargs and context result is None
+  - api_server_pillar: null
+    config_options: null
+    kwargs: {"kubeconfig": "my-kwargs-kubeconfig.conf"}
+    result: ['my-kwargs-kubeconfig.conf', null]
+
+get_version_info:
+  - kubeconfig: ['kubeconfig.conf', 'my_context']
+    api_client: "my_api_client"
+    raises: False
+    result: {
+      "major": "1",
+      "go_version": "go1.13.8",
+    }
+  - kubeconfig: ['kubeconfig.conf', 'my_context']
+    api_client: null
+    raises: True
+    result: 'Failed to get version info'
+
+read_and_render_yaml_file:
+  - source:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: kube-apiserver
+    template: null
+    context: null
+    raises: False
```

Comment:
Make test simpler to read please
Here you can keep the default value for all of them